integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -0,0 +1,243 @@
+from haystack.components.generators.utils import deserialize_callback_handler
+from haystack.dataclasses import ChatMessage, StreamingChunk
+
+from haystack_integrations.components.generators.amazon_bedrock.errors import (
+    AmazonBedrockConfigurationError,
+    AmazonBedrockInferenceError,
+    AWSConfigurationError,
+)
+
+from .adapters import AnthropicClaudeChatAdapter, BedrockModelChatAdapter, MetaLlama2ChatAdapter
+
+logger = logging.getLogger(__name__)
+
+AWS_CONFIGURATION_KEYS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_session_token",
+    "aws_region_name",
+    "aws_profile_name",
+]
+
+
+@component
+class AmazonBedrockChatGenerator:
+    """
+    AmazonBedrockChatGenerator enables text generation via Amazon Bedrock chat hosted models. For example, to use
+    the Anthropic Claude model, simply initialize the AmazonBedrockChatGenerator with the 'anthropic.claude-v2'
+    model name.
+
+    ```python
+    from haystack_integrations.components.generators.amazon_bedrock import AmazonBedrockChatGenerator
+    from haystack.dataclasses import ChatMessage
+    from haystack.components.generators.utils import print_streaming_chunk
+
+    messages = [ChatMessage.from_system("\\nYou are a helpful, respectful and honest assistant"),
+                ChatMessage.from_user("What's Natural Language Processing?")]
+
+
+    client = AmazonBedrockChatGenerator(model="anthropic.claude-v2", streaming_callback=print_streaming_chunk)
+    client.run(messages, generation_kwargs={"max_tokens_to_sample": 512})
+
+    ```
+
+    If you prefer non-streaming mode, simply remove the `streaming_callback` parameter, capture the return value of the
+    component's run method and the AmazonBedrockChatGenerator will return the response in a non-streaming mode.
+    """
+
+    SUPPORTED_MODEL_PATTERNS: ClassVar[Dict[str, Type[BedrockModelChatAdapter]]] = {
+        r"anthropic.claude.*": AnthropicClaudeChatAdapter,
+        r"meta.llama2.*": MetaLlama2ChatAdapter,
+    }
+
+    def __init__(
+        self,
+        model: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        stop_words: Optional[List[str]] = None,
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+    ):
+        """
+        Initializes the AmazonBedrockChatGenerator with the provided parameters. The parameters are passed to the
+        Amazon Bedrock client.
+
+        Note that the AWS credentials are not required if the AWS environment is configured correctly. These are loaded
+        automatically from the environment or the AWS configuration file and do not need to be provided explicitly via
+        the constructor.
+
```

Comment:
nit: from my experience, I suspect three minimum mandatory parameters need to be configured: `aws_access_key_id, aws_session_token, aws_region_name`, maybe a sentence in the docstring mention that